### PR TITLE
Prepared the what's new for v0.16.

### PR DESCRIPTION
**Pull request by pelson**
## Rationale

Docs for the v0.16.0 release so that users can quickly see what's changed.

## Implications

* Slightly improved the spacing on the docs (between class docs and paragraph sections)
* Properly documented the ``cartopy.feature.[A-Z]+`` features (they were previously documented as ``cartopy.feature.cartopy.feature.LAND``)
* Documented ``cartopy.feature.WFSFeature`` and ``NaturalEarthFeature.with_scale``
* Fixed a bug in the always_circular example introduced by #916.
* Added information for all examples in the gallery so that the tool-tip isn't empty.

### CODE REVIEW

File: docs/source/matplotlib/feature_interface.rst [hunk 1]
```diff
@@ -34,17 +37,29 @@ such as :func:`GeoAxes.add_feature <cartopy.mpl.geoaxes.GeoAxes.add_feature>`:
...
+.. py:data:: OCEAN                       Ocean polygons.
+
+.. py:data:: RIVERS                      Single-line drainages, including lake centerlines.
+
+.. py:data:: STATES                      Internal, first-order administrative boundaries
+                                         (limited to the United States at this scale).
+                                         Natural Earth have first-order admin boundaries
+                                         for most countries at 10m scale which may be
+                                         accessed with
+                                         ``cartopy.feature.STATES.with_scale('10m')``
```
Comment (QuLogic) [inline docs/source/matplotlib/feature_interface.rst]: You could remove some of the spaces in the first column and then maybe wrap this a bit less.

File: docs/source/whats_new.rst [hunk 2]
```diff
@@ -12,6 +12,85 @@ Features
...
 
+* The :ref:`gnomonic-projection` was brought up-to-date to include the
+  ``central_longitude`` argument. (:pull:`855`)
+
+* Ryan May improved the formulation of the boundary ellipse for the
+  :class:`~cartopy.crs.Geostationary` projection and added the
+  ``sweep_angle_axis`` keyword argument. (:pull:`890`, :pull:`897`) 
+
+* Elliott Sales de Andrade made a number of micro-optimisations to the
+  matplotlib interface, fixed a number of documentation issues with
```
Comment (QuLogic) [inline docs/source/whats_new.rst]: Matplotlib, in all places.

File: docs/source/whats_new.rst [hunk 3]
```diff
@@ -12,6 +12,85 @@ Features
...
+  python 3 and added matplotlib 2.0 & 2.1 compatibility. (:pull:`886`,
```
Comment (QuLogic) [inline docs/source/whats_new.rst]: Python

File: docs/source/whats_new.rst [hunk 4]
```diff
@@ -12,6 +12,85 @@ Features
...
+  :pull:`901`, :pull:`780`, :pull:`773`, :pull:`977`)
+
+* Tick padding was added to the to gridliner.
```
Comment (QuLogic) [inline docs/source/whats_new.rst]: s/to the to/to the/

File: docs/source/whats_new.rst [hunk 5]
```diff
@@ -12,6 +12,85 @@ Features
...
+
+.. figure:: _images/understanding_transform-6.png
+   :target: tutorials/understanding_transform.html
+   :align: center
+
+* Daniel Kirkham and Daryl Herzmann made significant improvements to the
+  stability of polygon transformation. The changes reduce the frequency
+  of messages such as
+  ``Unidentified problem with geometry, linestring being re-added`` and
+  ``Self-intersection at or near point <X> <Y>`` occuring.
```
Comment (QuLogic) [inline docs/source/whats_new.rst]: occurring

**Comment by pelson** [comment]
Thanks @QuLogic.

Once this goes green, I'm pretty happy to merge it. At which point, I'll tag ``v0.16``.

File: lib/cartopy/examples/hurricane_katrina.py [hunk 6]
```diff
@@ -2,6 +2,9 @@
 Hurricane Katrina
 -----------------
 
+This example uses the power of Shapely to illustrate states that are likely to
+have been significantly impacted by Hurricane Katrine.
```
Comment (QuLogic) [inline lib/cartopy/examples/hurricane_katrina.py]: Katrina

**Comment by pelson** [comment]
Thanks for the great review @QuLogic. Definitely a case of 2x 👀 are better than 1! 😄

**Pull request merged**